@navikt/aksel/src/codemod/transforms/darkside/box-to-boxnew/box-to-boxnew.ts

```diff
@@ -0,0 +1,61 @@
+import type { API, FileInfo } from "jscodeshift";
+import { legacyTokenConfig } from "../../../../darkside/config/legacy.tokens";
+import {
+  findComponentImport,
+  findJSXElement,
+  findProp,
+} from "../../../utils/ast";
+import { getLineTerminator } from "../../../utils/lineterminator";
+import moveAndRenameImport from "../../../utils/moveAndRenameImport";
+
+const propsAffected = ["background", "borderColor", "shadow"];
+
+export default function transformer(file: FileInfo, api: API) {
+  const j = api.jscodeshift;
+  const root = j(file.source);
+
+  const toSourceOptions = getLineTerminator(file.source);
+
+  const localName = findComponentImport({
+    file,
+    j,
+    name: "Box",
+    packageType: "react",
+  });
+
+  if (!localName) {
+    return;
+  }
+
+  const astElements = findJSXElement({
+    root,
+    j,
+    name: localName,
+    originalName: "Box",
+  });
+
+  for (const astElement of astElements.paths()) {
+    for (const prop of propsAffected) {
+      findProp({ j, path: astElement, name: prop }).forEach((attr) => {
+        const attrvalue = attr.value.value;
+        if (attrvalue.type === "StringLiteral") {
+          const config = legacyTokenConfig[attrvalue.value];
+          if (config?.replacement) {
+            attrvalue.value = config.replacement;
+          } else if (config?.comment) {
+            attrvalue.comments = [j.commentLine(config.comment)];
+          }
+        }
+      });
+    }
+  }
+
+  moveAndRenameImport(j, root, {
+    fromImport: "@navikt/ds-react",
+    toImport: "@navikt/ds-react",
+    fromName: "Box",
+    toName: "BoxNew",
+  });
+
```

Comment:
Migrating to `BoxNew` should only be necessary if you have used any of `propsAffected`. If you only have padding, margin etc `Box` still works with new theme.

Reply:
it's not nice to "upgrade" for the future then? 🤔 (i think it wouldn't hurt, but maybe it's more mental overhead than required... at least in the short term?), but the more i think about this, the better it is to switch over early, and all the better if there's no change in behaviour?

Reply:
there's always the chance that you start to diverge from easily migrated box into one that has no analogue in the new system.

Reply:
When the `darkside` is set as the default, we will delete `BoxNew` and just use `Box`, so `BoxNew` is mainly just for those testing the new system. Bu i guess you're right, it wont hurt to just migrate to `BoxNew` always to avoid having both of them used at the same time 👍